backend/app/receive.py

```diff
@@ -0,0 +1,61 @@
+import datetime
+import os.path
+import uuid
+
+from flask import request, abort
+import psycopg2
+from webargs import fields
+from webargs.flaskparser import use_args
+
+from . import app
+from .authorize_station import authorize_station
+from .utils import make_thumbnail
+
+cfg = app.config["database"]
+
+@app.route('/receive', methods=["POST",])
+@authorize_station
+@use_args({
+    'file': fields.Field(validate=lambda file: file.mimetype == "image/png", location="files"),
+    'aos': fields.DateTime(required=True),
+    'tca': fields.DateTime(required=True),
+    'los': fields.DateTime(required=True),
+    'sat': fields.Str(required=True),
+    'notes': fields.Str(required=False)
+})
+def receive(station_id, args):
+    if len(request.files) == 0:
+        abort(400)
+
+    file_ = args['file']
+    filename = "%s-%s" % (str(uuid.uuid4()), file_.filename)
+
+    with psycopg2.connect(**cfg) as conn:
+        with conn.cursor() as cursor:
+            cursor.execute("SELECT sat_id FROM satellites WHERE sat_name = %s LIMIT 1;", (args["sat"],))
+            row = cursor.fetchone()
+            if row is None:
+                abort(400)
```

Comment:
The same as above - error information should be communicated back. Something like "No satellite XYZ in the database.". Also, I think the logic here is broken. In the future the system should accept observations even for unknown sats. So I'd use sat_id -1 or 0 to designate unknown sat and just accept the observation.

Reply:
Fixed